### PR TITLE
Add a quickwit-integration-tests crate

### LINKED ISSUE #2984: Add a quickwit-integration-tests crate
We started a bunch of "e2e" tests [there](https://github.com/quickwit-oss/quickwit/blob/main/quickwit/quickwit-serve/src/tests.rs)

As we need to do more of them to cover the basic actions a user will do when using quickwit, we should probably isolate them and use the `quickwit-rest-client` for that.

Thus `quickwit-integration-tests`.

### PR DESCRIPTION
**Pull request by imotov**
### Description

Fixes #2984

### How was this PR tested?

The whole PR is basically a test infrastructure.

### CODE REVIEW

File: quickwit/quickwit-integration-tests/src/test_utils/test_client.rs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    pub fn new(addr: SocketAddr) -> Self {
+        let client = hyper::Client::builder()
+            .pool_idle_timeout(Duration::from_secs(30))
+            .http2_only(true)
+            .build_http();
+        let root_url = format!("http://{addr}");
+        Self { root_url, client }
+    }
+
+    pub async fn is_live(&self) -> anyhow::Result<bool> {
```
Comment (fmassot) [inline quickwit/quickwit-integration-tests/src/test_utils/test_client.rs:42]: Is there a particular reason to not put these endpoints in the rest client?

Comment (imotov) [reply quickwit/quickwit-integration-tests/src/test_utils/test_client.rs:42]: I had a reason, but now I see it was a wrong reason 😄 . Somehow, I thought these endpoints are enabled only during a test run, but now I see that I misinterpreted this part. I will fix that.

File: quickwit/quickwit-cli/src/service.rs [hunk 2]
```diff
@@ -80,7 +81,12 @@ impl RunCliCommand {
         quickwit_telemetry::send_telemetry_event(telemetry_event).await;
         // TODO move in serve quickwit?
         start_actor_runtimes(&config.enabled_services)?;
-        serve_quickwit(config).await?;
+        let _ = serve_quickwit(config, async move {
```
Comment (fmassot) [inline quickwit/quickwit-cli/src/service.rs:84]: I think we should take into account other signals (I had this [kind of stuff in mind](https://github.com/vectordotdev/vector/blob/68b54a9bc0ae07d916ec48e997a03f7681e54ccc/src/signal.rs#L44)
I suggest to do that in another PR so that you can advance on your original issue :)

Comment (fmassot) [reply quickwit/quickwit-cli/src/service.rs:84]: There is an issue related to this https://github.com/quickwit-oss/quickwit/issues/3073

**Review by fmassot — APPROVED** [review]
We now have this integration tests project, that's really cool, thanks for taking care of this issue. I want to review the shutdown signals but let's do it in another PR.
